node/pkg/reporter/main_test.go

```diff
@@ -2,172 +2,69 @@
 package reporter
 
 import (
-	"bytes"
 	"context"
-	"encoding/json"
 	"os"
 	"testing"
-	"time"
 
-	"bisonai.com/orakl/node/pkg/admin/reporter"
-	"bisonai.com/orakl/node/pkg/admin/utils"
-	"bisonai.com/orakl/node/pkg/bus"
-	"bisonai.com/orakl/node/pkg/chain/helper"
-	"bisonai.com/orakl/node/pkg/common/keys"
 	"bisonai.com/orakl/node/pkg/db"
-	"github.com/gofiber/fiber/v2"
+	"bisonai.com/orakl/node/pkg/utils/request"
 	"github.com/rs/zerolog"
 	"github.com/rs/zerolog/log"
 )
 
-const InsertGlobalAggregateQuery = `INSERT INTO global_aggregates (config_id, value, round, timestamp) VALUES (@config_id, @value, @round, @timestamp) RETURNING *`
-const InsertConfigQuery = `INSERT INTO configs (name, fetch_interval, aggregate_interval, submit_interval) VALUES (@name, @fetch_interval, @aggregate_interval, @submit_interval) RETURNING name, id, submit_interval, aggregate_interval;`
-const TestInterval = 15000
+const InsertConfigQuery = `INSERT INTO configs (name, aggregate_interval, submit_interval) VALUES (@name, @aggregate_interval, @submit_interval) RETURNING name, id, submit_interval, aggregate_interval;`
 
-type TestItems struct {
-	app        *App
-	admin      *fiber.App
-	messageBus *bus.MessageBus
-	tmpData    *TmpData
-}
-type TmpData struct {
-	globalAggregate GlobalAggregate
-	config          Config
-	proofBytes      []byte
-	proofTime       time.Time
-}
-
-func insertSampleData(ctx context.Context) (*TmpData, error) {
-	var tmpData = new(TmpData)
-
-	signHelper, err := helper.NewSigner(ctx)
-	if err != nil {
-		return nil, err
-	}
-	proofTime := time.Now()
-
-	tmpConfig, err := db.QueryRow[Config](ctx, InsertConfigQuery, map[string]any{"name": "test-aggregate", "submit_interval": TestInterval, "fetch_interval": TestInterval, "aggregate_interval": TestInterval})
-	if err != nil {
-		log.Error().Err(err).Msg("error inserting config 0")
-		return nil, err
-	}
-	tmpData.config = tmpConfig
-
-	err = db.QueryWithoutResult(ctx, InsertConfigQuery, map[string]any{"name": "test-aggregate-2", "submit_interval": TestInterval * 2, "fetch_interval": TestInterval, "aggregate_interval": TestInterval})
-	if err != nil {
-		log.Error().Err(err).Msg("error inserting config 1")
-		return nil, err
+func getConfigUrl() string {
+	configUrl := os.Getenv("ADMIN_ENDPOINT")
+	if configUrl == "" {
+		configUrl = "http://100.65.92.37:3030/api/v1"
```

Comment:
it won't probably work in github action since admin endpoint is only accessible within k8s cluster or tailscale

Reply:
hmm so we don't have a way to access the admin endpoint without tailscale? 😨

Reply:
yes, but there are few options

- run admin service locally from reporter test and use it
- access orakl.config to get configs
- insert dummy config data

Reply:
yep, went with the third option :) thanx for the suggestions